From 5f5213d4320b4ca2263b1210969241534dfc8ba3 Mon Sep 17 00:00:00 2001
From: Dorian Stoll <dorian.stoll@tmsp.io>
Date: Sat, 22 Jul 2023 17:08:56 +0200
Subject: [PATCH 34/62] Update IPTS from module repo

Changes:
 * Fix allocating the HID descriptor on EDS v2

Idk why I wrote that, for EDS v1 I got it right ...

Based on https://github.com/linux-surface/intel-precise-touch/commit/32622a37f27113067a6c9aff21627859caab43f9

Signed-off-by: Dorian Stoll <dorian.stoll@tmsp.io>
Patchset: ipts
---
 drivers/hid/ipts/eds2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hid/ipts/eds2.c b/drivers/hid/ipts/eds2.c
index e835b460aa79..7c5cbafdb279 100644
--- a/drivers/hid/ipts/eds2.c
+++ b/drivers/hid/ipts/eds2.c
@@ -34,7 +34,7 @@ int ipts_eds2_get_descriptor(struct ipts_context *ipts, u8 **desc_buffer, size_t
 	size = sizeof(ipts_singletouch_descriptor) + ipts->descriptor.size;
 
 	buffer = kzalloc(size, GFP_KERNEL);
-	if (!*buffer)
+	if (!buffer)
 		return -ENOMEM;
 
 	memcpy(buffer, ipts_singletouch_descriptor, sizeof(ipts_singletouch_descriptor));
-- 
2.39.1

